src/libraries/Directory.Build.props

```diff
@@ -103,6 +103,14 @@
     <IsPackable Condition="$(MSBuildProjectName.StartsWith('Microsoft.Extensions.')) and '$(IsSourceProject)' == 'true'">true</IsPackable>
   </PropertyGroup>
 
+  <Import Sdk="Microsoft.DotNet.PackageValidation" Project="Sdk.props" Condition="'$(IsPackable)' == 'true' and '$(IsSourceProject)' == 'true'"/>
```

Comment:
Why do you need/want the IsSourceProject property? Should we instead condition it on Extension != ".pkgproj"?

Reply:
It would be better to save this condition as a property and use it here and for the property group.